Core/src/org/sleuthkit/autopsy/communications/relationships/CallLogNode.java

```diff
@@ -107,6 +100,59 @@ long getCallDuration(BlackboardArtifact artifact) throws TskCoreException {
         return endAttribute.getValueLong() - startAttribute.getValueLong();
     }
     
+    /**
+     * Returns the phone number to display in the To/From column. The number is
+     * picked from one of the 3 possible phone number attributes, based on the
+     * direction of the call.
+     *
+     * @param artifact Call log artifact.
+     *
+     * @return Phone number to display.
+     */
+    private String getPhoneNumber(BlackboardArtifact artifact) {
+        String direction = getAttributeDisplayString(artifact, TSK_DIRECTION);
+
+        String phoneNumberToReturn;
+        String fromPhoneNumber = getAttributeDisplayString(artifact, TSK_PHONE_NUMBER_FROM);
+        String toPhoneNumber = getAttributeDisplayString(artifact, TSK_PHONE_NUMBER_TO);
+        String phoneNumber = getAttributeDisplayString(artifact, TSK_PHONE_NUMBER);
+        switch (direction.toLowerCase()) {
+            case "incoming": // NON-NLS 
+                phoneNumberToReturn = getFirstNonBlank(fromPhoneNumber, phoneNumber, toPhoneNumber);
+                break;
+            case "outgoing": // NON-NLS
+                phoneNumberToReturn = getFirstNonBlank(toPhoneNumber, phoneNumber, fromPhoneNumber);
+                break;
+            default:
+                phoneNumberToReturn = getFirstNonBlank(toPhoneNumber, fromPhoneNumber, phoneNumber );
+                break;
+        }
+
+        return phoneNumberToReturn;
+    }
+    
+    /**
+     * Checks the given string arguments in order and returns the first non blank string.
+     * Returns a blank string if all the input strings are blank.
+     * 
+     * @param string1 First string to check
+     * @param string2 Second string to check
+     * @param string3 Third string to check
+     * 
+     * @retunr first non blank string if there is one, blank string otherwise.
+     * 
+     */
+    private String getFirstNonBlank(String string1, String string2, String string3 ) {
```

Comment:
Coalescing blankness is a very common operation. Since you are using commons lang, they've already implemented this for you. 

https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#firstNonBlank-T...-

I suggest replacing the above getFirstNonBlank calls with StringUtils.firstNonBlank(...) and then, before returning, check if phoneNumberToReturn is null. If so, you can sub an empty value like you're doing here.

Reply:
The firstNonBlank() method in StringUtils is available from version 3.8 of commons-lang3.

We are currently using 3.0 (but Thunderbird uses 3.8.1)

@rcordovano: is it ok to update the commons-lang3 to latest ?

Reply:
Yes.